### PR TITLE
Support for self update in AY mode

### PR DESCRIPTION
**Pull request by lslezak**
Extends #352, #356 also to AutoYaST mode.

See [FATE#319716](https://fate.suse.com/319716).

The AY profile schema is updated in https://github.com/yast/yast-autoinstallation/pull/206.

### CODE REVIEW

File: src/lib/installation/clients/inst_update_installer.rb [hunk 1]
```diff
@@ -83,7 +84,8 @@ def self_update_enabled?
     # @see #self_update_url_from_linuxrc
     # @see #self_update_url_from_control
```
Comment (imobach) [inline src/lib/installation/clients/inst_update_installer.rb]: np: @see #self_update_url_from_profile

File: src/lib/installation/clients/inst_update_installer.rb [hunk 2]
```diff
@@ -101,6 +103,19 @@ def self_update_url_from_control
       get_url_from(ProductFeatures.GetStringFeature("globals", "self_update_url"))
     end
 
+    # Return the self-update URL from the AutoYaST profile
+    # @return [URI,nil] the self-update URL, nil if not running in AutoYaST mode
+    #   or when the URL is not defined in the profile
+    def self_update_url_from_profile
+      return nil unless Mode.auto
+
+      Yast.import "Profile"
```
Comment (imobach) [inline src/lib/installation/clients/inst_update_installer.rb]: Why not move this to `main`?

Comment (lslezak) [inline src/lib/installation/clients/inst_update_installer.rb]: The reason is that the `Mode` is always needed (to know whether we are running in AY or not), but `Profile` is needed only in AY mode, in normal installation it's not needed.

(Maybe the `Profile` is loaded by other code anyway and this small optimization is useless, but at least I tried... :wink:)

File: test/test_helper.rb [hunk 3]
```diff
@@ -11,10 +11,20 @@ module Yast
   class AutoinstConfigClass
     # we need at least one non-default methods, otherwise ruby-bindings thinks
     # it is just namespace
-    def fake_method
+    def cio_ignore
```
Comment (imobach) [inline test/test_helper.rb]: Uh, I didn't know about this.

**Comment by imobach** [comment]
LGTM

**Pull request merged**